src/client/profile-provider.ts

```diff
@@ -97,6 +101,10 @@ export class BoundProfileProvider {
     input?: TInput
   ): Promise<Result<TResult, ProfileParameterError | MapInterpreterError>> {
     // compose and validate the input
+    Reflect.set(this.fetchInstance, 'metadata', {
+      profile: profileAstId(this.profileAst),
+      usecase,
+    });
```

Comment:
Why is this needed?

Reply:
to let fetch interceptors know which usecase and profile you're calling